caikit/interfaces/nlp/data_model/classification.py

```diff
@@ -0,0 +1,136 @@
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Data structures for classification representations"""
+
+# Standard
+from typing import List, Optional
+
+# Third Party
+import numpy as np
+
+# First Party
+from py_to_proto.dataclass_to_proto import Annotated, FieldNumber
+import alog
+
+# Local
+from ....core import DataObjectBase, dataobject
+from .package import NLP_PACKAGE
+from .text_generation import FinishReason
+
+log = alog.use_channel("DATAM")
+
+
+@dataobject(package=NLP_PACKAGE)
+class ClassificationTrainRecord(DataObjectBase):
+    """A classification training record consisting of a single train instance."""
+
+    text: Annotated[str, FieldNumber(1)]  # Text to be classified
+    labels: Annotated[
+        List[str], FieldNumber(2)
+    ]  # Class labels to be learnt for the text
+
+
+@dataobject(package=NLP_PACKAGE)
+class Classification(DataObjectBase):
+    """A single classification prediction."""
+
+    label: Annotated[str, FieldNumber(1)]  # Predicted relevant class name
+    score: Annotated[
+        float, FieldNumber(2)
+    ]  # The confidence-like score of this prediction in [0, 1]
+
+
+@dataobject(package=NLP_PACKAGE)
+class ClassificationResult(DataObjectBase):
+    """Classification result generated from a text and consisting multiple classes."""
+
+    results: Annotated[
+        List[Classification], FieldNumber(1)
+    ]  # List of classifications for a text
+
+
+# NOTE: This is meant to align with the HuggingFace token classification task:
+# https://huggingface.co/docs/transformers/tasks/token_classification#inference
+# The field `word` does not necessarily correspond to a single "word",
+# and `entity` may not always be applicable beyond "entity" in the NER
+# (named entity recognition) sense
+@dataobject(package=NLP_PACKAGE)
+class TokenClassification(DataObjectBase):
+    """A single token classification prediction."""
+
+    start: Annotated[int, FieldNumber(1)]  # Beginning offset of the token
+    end: Annotated[int, FieldNumber(2)]  # Ending offset of the token
+    word: Annotated[str, FieldNumber(3)]  # Text referenced by this token
+    entity: Annotated[
+        str, FieldNumber(4)
+    ]  # Predicted relevant class name for the token
+    entity_group: Annotated[str, FieldNumber(5)]  # Aggregate label, if applicable
+    score: Annotated[
+        float, FieldNumber(6)
+    ]  # The confidence-like score of this classification prediction in [0, 1]
+    token_count: Annotated[
+        Optional[int], FieldNumber(7)
+    ]  # Length of tokens in the text
+
+
+@dataobject(package=NLP_PACKAGE)
+class TokenClassificationResult(DataObjectBase):
+    """Token classification result generated from a text and consisting multiple classes."""
+
+    results: Annotated[List[TokenClassification], FieldNumber(1)]
+
+
+@dataobject(package=NLP_PACKAGE)
+class TokenClassificationStreamResult(TokenClassificationResult):
```

Comment:
The name of the datamodel seems off, since it seems to be only returning processed_index and not the "token classification" itself ?

Reply:
oh make sense. thanks